### ISSUE TITLE
Not able to build VM image on Ubuntu 20.04 from master branch

### ISSUE THREAD

**Issue by gmshastri** [sonic-net/sonic-buildimage#17485]
Hi Team,

My host machine is "Ubuntu 20.04"

$ cat /etc/lsb-release 

DISTRIB_ID=Ubuntu

DISTRIB_RELEASE=20.04

DISTRIB_CODENAME=focal

DISTRIB_DESCRIPTION="Ubuntu 20.04 LTS"

I am trying to build VM image using following steps:
==
1. Referring this page : https://github.com/sonic-net/sonic-buildimage/tree/master#build-sonic-switch-images
2. Install all the required docker packages supporting Ubuntu 20.04 as per above page guideline.
3. modprobe overlay
4. git clone --recurse-submodules https://github.com/sonic-net/sonic-buildimage.git
5. cd sonic-buildimage
6. make init
7. make configure PLATFORM=vs
end up with following errors:

Step 9/58 : COPY ["buildinfo", "/usr/local/share/buildinfo"]                                                      

 ---> 3e48a492ca8b                                                                                                

Step 10/58 : COPY vcache/ /sonic/target/vcache/sonic-slave-bookworm                                               

 ---> 70e57744341b                                                                                                

Step 11/58 : RUN dpkg -i /usr/local/share/buildinfo/sonic-build-hooks_1.0_all.deb                                 

 ---> Running in f9740e48f605                                                                                     

dpkg-deb (subprocess): decompressing archive '/usr/local/share/buildinfo/sonic-build-hooks_1.0_all.deb' (size=1262

0) member 'control.tar': lzma error: Cannot allocate memory                                                       

tar: This does not look like a tar archive                                                                        

tar: Exiting with failure status due to previous errors                                                           

dpkg-deb: error: tar subprocess returned error exit status 2                                                      

dpkg: error processing archive /usr/local/share/buildinfo/sonic-build-hooks_1.0_all.deb (--install):              

 dpkg-deb --control subprocess returned error exit status 2                                                       

Errors were encountered while processing:                                                                         

 /usr/local/share/buildinfo/sonic-build-hooks_1.0_all.deb                                                         

The command '/bin/sh -c dpkg -i /usr/local/share/buildinfo/sonic-build-hooks_1.0_all.deb' returned a non-zero code

: 1                                                                                                               

Unable to find image 'sonic-slave-bookworm:cf925688f01' locally                                                   

Error response from daemon: pull access denied for sonic-slave-bookworm, repository does not exist or may require 

'docker login': denied: requested access to the resource is denied

Error: No such container:path: sonic-slave-bookworm-cf925688f01:/etc/os-release

Error: No such container:path: sonic-slave-bookworm-cf925688f01:/usr/local/share/buildinfo/pre-versions

Error: No such container:path: sonic-slave-bookworm-cf925688f01:/usr/local/share/buildinfo/post-versions

Error response from daemon: No such image: sonic-slave-bookworm:cf925688f01

[+] Building 1.2s (3/4)     

 => [internal] load .dockerignore                                                                            0.5s

 => => transferring context: 2B                                                                              0.0s

 => [internal] load build definition from Dockerfile.cleanup                                                 0.6s

 => => transferring dockerfile: 326B                                                                         0.0s

 => ERROR FROM [docker.io/library/sonic-slave-bookworm:cf925688f01](http://docker.io/library/sonic-slave-bookworm:cf925688f01)                                            0.6s

 => => resolve [docker.io/library/sonic-slave-bookworm:cf925688f01](http://docker.io/library/sonic-slave-bookworm:cf925688f01)                                            0.6s

------

 > FROM [docker.io/library/sonic-slave-bookworm:cf925688f01](http://docker.io/library/sonic-slave-bookworm:cf925688f01):

------

failed to solve with frontend dockerfile.v0: failed to build LLB: failed to load cache key: pull access denied, re

pository does not exist or may require authorization: server message: insufficient_scope: authorization failed

[+] Building 1.1s (4/5)     

 => [internal] load build definition from Dockerfile.cleanup                                                 0.5s

 => => transferring dockerfile: 48B                                                                          0.0s

 => [internal] load .dockerignore                                                                            0.3s

 => => transferring context: 2B                                                                              0.0s

 => ERROR [internal] load metadata for [docker.io/library/sonic-slave-bookworm:cf925688f01](http://docker.io/library/sonic-slave-bookworm:cf925688f01)                    0.2s

 => ERROR [final 1/2] FROM [docker.io/library/sonic-slave-bookworm:cf925688f01](http://docker.io/library/sonic-slave-bookworm:cf925688f01)                                0.2s

 => => resolve [docker.io/library/sonic-slave-bookworm:cf925688f01](http://docker.io/library/sonic-slave-bookworm:cf925688f01)                                            0.2s

------

 > [internal] load metadata for [docker.io/library/sonic-slave-bookworm:cf925688f01](http://docker.io/library/sonic-slave-bookworm:cf925688f01):

------

------

 > [final 1/2] FROM [docker.io/library/sonic-slave-bookworm:cf925688f01](http://docker.io/library/sonic-slave-bookworm:cf925688f01):

 

------

failed to solve with frontend dockerfile.v0: failed to build LLB: failed to load cache key: pull access denied, re

pository does not exist or may require authorization: server message: insufficient_scope: authorization failed

Error: No such image: tmp-sonic-slave-bookworm:cf925688f01

Error: No such container:path: sonic-slave-bookworm-cf925688f01:/usr/local/share/buildinfo/log

sync: error opening 'sonic-slave-bookworm/vcache/cache.base64': No such file or directory

Error: No such container: sonic-slave-bookworm-cf925688f01

make[1]: *** [Makefile.work:624: configure] Error 1

make[1]: Leaving directory '/var/gshastri/sonic-buildimage'

 

make: *** [Makefile:109: configure] Error 2

Would you please let me know what could be the issue here? Is there any specific branch for "vs" target? 

My final command after above command will be,
"make SONIC_BUILD_JOBS=4 target/sonic-vs.bin"

**Comment by hasan-alkama** [comment]
Can you share the full error log,
is it able to build the bullseye and buster image? during configure step

**Comment by gmshastri** [comment]
Hi @hasan-alkama 

I have not set any environment variable like NOBULLSEYE , NOBUSTER, etc. 
I am just simply trying to build VS image from master branch.

As you see from above, error comes at "make configure PLATFORM=vs" state only.

**Comment by hasan-alkama** [comment]
Yes you Actually no need to do that it is already there inside Makefile.
The thing is I was also trying to build multiarch qemu vm image for sonic-vs 
and getting error in bookworm then I changed the flag inside Makefile to NOBOOKWORM=1
as i don't want to build the bookworm (as the multiarch/debian-debootstrap arm64-bookworm is not available)

I am not sure but you can also try once without bookworm

**Comment by prabhataravind** [comment]
@gmshastri could you please check your build environment and try with the latest SHA on master?

**Comment by prabhataravind** [comment]
Please work with @xumia if the issue still persists.

**Comment by saiarcot895** [comment]
This is the actual error message:

```
dpkg-deb (subprocess): decompressing archive '/usr/local/share/buildinfo/sonic-build-hooks_1.0_all.deb' (size=1262
member 'control.tar': lzma error: Cannot allocate memory
```

This is due to an issue between the kernel running on the host machine and the Docker daemon version that is used. More details are in #16789. The fix is to upgrade Docker to 20.10.10 or newer.

What version of Docker are you using? Right now, I haven't added a hard requirement on what version of Docker should be used, since I wasn't entirely clear on what versions are actually affected here.
